interceptor.go

```diff
@@ -204,18 +205,28 @@ func (i *Interceptor) WrapStreamingClient(next connect.StreamingClientFunc) conn
 			instrumentation.requestSize,
 			instrumentation.requestsPerRPC,
 		)
-		ctx, span := i.config.tracer.Start(
-			ctx,
-			name,
-			trace.WithSpanKind(trace.SpanKindClient),
-			trace.WithAttributes(state.attributes...),
-		)
-		// inject the newly created span into the carrier
-		carrier := propagation.HeaderCarrier(conn.RequestHeader())
-		i.config.propagator.Inject(ctx, carrier)
+		var span trace.Span
+		var createSpanOnce sync.Once
+		createSpan := func() {
+			ctx, span = i.config.tracer.Start(
+				ctx,
+				name,
+				trace.WithSpanKind(trace.SpanKindClient),
+				trace.WithAttributes(state.attributes...),
+				trace.WithAttributes(headerAttributes(
+					protocol,
+					requestKey,
+					conn.RequestHeader(),
+					i.config.requestHeaderKeys)...),
+			)
+			// inject the newly created span into the carrier
+			carrier := propagation.HeaderCarrier(conn.RequestHeader())
+			i.config.propagator.Inject(ctx, carrier)
+		}
 		return &streamingClientInterceptor{
 			StreamingClientConn: conn,
 			onClose: func() {
+				createSpanOnce.Do(createSpan)
```

Comment:
It doesn't look to me like this is thread-safe. `onClose` is called from `CloseResponse`, which must be safe to call concurrently with `Send`. (This is documented on the `StreamingClientConn` interface.) Using a `sync.Once` guarantees that `createSpan` is only called once, but it does _not_ guarantee that the span has actually been created - it's possible for a call to `CloseResponse` to be in the midst of creating the span while a concurrent call to `Send` skips over the `Once` and attempts to use the nonexistent span.

Let's pair program this afternoon and try to fix this up!